docs/products/opensearch/howto/upgrade-to-opensearch.rst

```diff
@@ -43,3 +43,58 @@ Extra information for Terraform users
 -------------------------------------
 
 If you manage your infrastructure with our :doc:`Terraform provider </docs/tools/terraform>` then please note that you will need to upgrade the provider to the latest release in order to have support for OpenSearch.
+
+To perform the upgrade in place using Terraform, you will need to take the following steps:
+
+1. Change the ``elasticsearch_version = 7`` to ``opensearch_version = 1``. This is the equivalent to clicking the migrate button in the console.
+
+.. code-block::
+
+    resource "aiven_elasticsearch" "es" {
+      project = "project-name"
+      cloud_name = "google-us-east4"
+      plan = "business-4"
+      service_name = "es"
+
+      elasticsearch_user_config {
+        elasticsearch_version = 7
+      }
+    }
+
+.. code-block::
+
+    resource "aiven_elasticsearch" "es" {
+      project = "project-name"
+      cloud_name = "google-us-east4"
+      plan = "business-4"
+      service_name = "es"
+
+      elasticsearch_user_config {
+        opensearch_version = 1
+      }
+    }
+
+2. After the migration you will need to remove the Elasticsearch service from the Terraform state.
+
+.. code-block::
+
+    terraform state rm 'aiven_elasticsearch.es'
+
+3. Add the OpenSearch service to the Terraform state
+
+.. code-block::
+
+    resource "aiven_opensearch" "os" {
```

Comment:
I'm also a bit confused why the change above is not equal to this one. Is it that we need to change elasticsearch_version to opensearch_version in one step and only then update name to import it to terraform state?

Reply:
Yes, these steps must be taken separately and linearly. If not TF will not be able to migrate properly.